spec/inputs/http_poller_spec.rb

```diff
@@ -23,7 +23,11 @@
   }
   let(:klass) { LogStash::Inputs::HTTP_Poller }
 
-  subject { klass.new(default_opts) }
+  subject {
+    s = klass.new(default_opts)
+    s.register # We can't use a before block here because of rspec scoping
```

Comment:
really? What happens if we do use a before? (I typically use `before { plugin.register }` without issue, but maybe this case has issues?